cdci_data_analysis/analysis/instrument.py

```diff
@@ -253,6 +255,32 @@ def parse_inputs_files(self,
 
             raise RequestNotUnderstood(error_message)
 
+    def get_status_details(self,
+                           par_dic,
+                           config=None,
+                           logger=None):
+        if logger is None:
+            logger = self.get_logger()
+
+        status_details = None
+
+        # TODO put this in a dedicated function, perhaps within the oda_api
+        # adaptation for oda_api, like it happens in oda_api set_api_code function
+        updated_par_dic = par_dic.copy()
+        updated_par_dic['product'] = updated_par_dic['product_type']
+        updated_par_dic['product_type'] = updated_par_dic['query_type']
+        updated_par_dic.pop('query_type')
+
+        logger.info(f"getting products for a more in-depth analysis for the results within run_call_back with args {updated_par_dic}")
+        disp = DispatcherAPI(url=config.dispatcher_callback_url_base, instrument='mock')
+        try:
+            disp.get_product(**updated_par_dic)
+        except RemoteException as re:
+            logger.info('RemoteException detected when retrieving additional information from a completed job')
```

Comment:
> I think this is the case when we do not need to send an email, since it will not give user anything actionable. Instead, we should make a sentry report.
> 

So we should include within the email any error details/message? Returned from this call I mean.

>
> Except maybe if this is returned in case of, say, empty result. This should be sent to user with an appropriate, careful, comment. Could you test this empty result case?
>

How would we detect an empty result?

Reply:
> > I think this is the case when we do not need to send an email, since it will not give user anything actionable. Instead, we should make a sentry report.
> 
> So we should include within the email any error details/message? Returned from this call I mean.

Yes, see example email in test.

> 
> > Except maybe if this is returned in case of, say, empty result. This should be sent to user with an appropriate, careful, comment. Could you test this empty result case?
> 
> How would we detect an empty result?

Check what oda_api raises. Sometimes it's connection error - this should not be sent. Everything that successfully returns from backend but raises an exception in processing is effectivly empty.

Reply:
> 
> Check what oda_api raises. Sometimes it's connection error - this should not be sent. Everything that successfully returns from backend but raises an exception in processing is effectivly empty.
>

What do you mean with processing? Processing, by the oda_api, of what has been returned?
There are several exception raised by oda_api, and most of them derive from `RemoteException`

Reply:
can you distinguish connection error?

Reply:
Yes that can be done. However also other exceptions generated within the oda_api can be intercepted, namely:

- `DispatcherNotAvailable`

Reply:
Where can I find an example of an empty product? I am trying to design a fake product that would indeed return an empty product for tests

Reply:
You can put a fake ScW ID:

```bash
$ oda-api get -i isgri -p isgri_image -a scw_list=351700110010.000 -a integral_data_rights=all-private
```

Reply:
SO I tried to intercept (almost) all the different sort of exceptions raised by the `oda_api`, until a RemoteException is detected that could possibly indicate an `empty_product`